android/app/src/main/python/mmrelay/matrix_utils.py

```
@@ -0,0 +1,2322 @@
+        # Attempt to join the room if not already joined
+        if room_id not in matrix_client.rooms:
+            response = await matrix_client.join(room_id)
+            if response and hasattr(response, "room_id"):
+                logger.info(f"Joined room '{room_id_or_alias}' successfully")
+            else:
+                logger.error(
+                    f"Failed to join room '{room_id_or_alias}': {getattr(response, 'message', str(response))}"
+                )
+        else:
+            logger.debug(f"Bot is already in room '{room_id_or_alias}'")
+    except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)